sklearn/ensemble/iforest.py

```diff
@@ -120,6 +120,10 @@ class IsolationForest(BaseBagging, OutlierMixin):
     verbose : int, optional (default=0)
         Controls the verbosity of the tree building process.
 
+    warm_start : bool, optional (default=False)
+        When set to ``True``, reuse the solution of the previous call to fit
+        and add more estimators to the ensemble, otherwise, just fit a whole
+        new forest. See :term:`the Glossary <warm_start>`.
```

Comment:
do we wanna say which parameters and in which order `warm_start` would be appropriate?

The test only checks for the `n_estimators`, for instance, what happens if we
`warm_start` on a larger value of `max_features`. I think we better clarify that
here (and maybe in the other ones?)

Reply:
When you say the other ones you mean the other bagging estimators such as `BaggingClassifier` or `RandomForest`? Also I don't know if this is the intended behavior of `warm_start` but I noticed here that it is possible to predict after updating `n_estimators` but without refitting before. This can be confusing for the users as `clf.n_estimators` is not the number used when predicting.

Reply:
I'm not sure what exactly you mean here. But (with `warm_start=True`) `fit()` will add new trees only if `n_estimators` is increased. Parameters of these new `ExtraTreeRegressor` trees will be set again using (possibly amended) parameters taking into account the size of to-be-fitted matrix. Don't know how it is elsewhere (although it seems to be similar in `BaseForest`).

Would you mean to mention this in the `warm_start` explanation, or do you have something else in mind? 

The tests only tests that new trees are indeed added. Would you want to modify that?

Reply:
> Also I don't know if this is the intended behavior of `warm_start` but I noticed here that it is possible to predict after updating `n_estimators` but without refitting before. This can be confusing for the users as `clf.n_estimators` is not the number used when predicting.

The behavior is the same for other scikit-learn estimators so never mind what I said :)

Reply:
> (with `warm_start=True`) `fit()` will add new trees only if `n_estimators` is increased. Parameters of these new `ExtraTreeRegressor` trees will be set again using (possibly amended) parameters taking into account the size of to-be-fitted matrix

It might be good to add something along these lines in the description of the `warm_start` parameter. Is that what you had in mind @adrinjalali?

Reply:
IMO it's pretty clear from the docstring that warm_start affects `n_estimators`, and the link to the docstring is also helpful

Reply:
as discussed below, a short explanation and example added to `outlier_detection.rst`